File: DSA/Java/src/dsa/lib/algo/sort/countingSort/CountingSort.java
package dsa.lib.algo.sort.countingSort;

import dsa.lib.utils.Range;

public class CountingSort {
    public static void sort(int[] nums) {
        Integer[] a = Range.integerValueOf(nums);
        new CountingSortInteger(a).sort();
        Range.copy(a, 0, nums, 0, nums.length);
    }

    public static void sort(Integer[] nums) {
        new CountingSortInteger(nums).sort();
    }

    public static void sort(int[] nums, int place) {
        Integer[] a = Range.integerValueOf(nums);
        new CountingSortIntegerPlace(a, place).sort();
        Range.copy(a, 0, nums, 0, nums.length);
    }

    public static void sort(Integer[] nums, int place) {
        new CountingSortIntegerPlace(nums, place).sort();
    }

    public static void sort(String[] nums, int index) {
        new CountingSortString(nums, index).sort();
    }
}

File: DSA/Python/src/dsa/lib/algo/sort/counting_sort/counting_sort.py
from typing import List, AnyStr

from dsa.lib.algo.sort.counting_sort.counting_sort_int import CountingSortInt, CountingSortIntPlace
from dsa.lib.algo.sort.counting_sort.counting_sort_str import CountingSortStr


def sortInt(nums: List[int]) -> None:
    CountingSortInt(nums).sort()


def sortIntPlace(nums: List[int], place: int) -> None:
    CountingSortIntPlace(nums, place).sort()


def sortStr(ss: List[AnyStr], index: int) -> None:
    CountingSortStr(ss, index).sort()
